src/main/java/org/wildfly/naming/client/WildFlyRootContext.java

```diff
@@ -98,9 +98,6 @@ public NameParser getNativeNameParser() {
     }
 
     protected Object lookupNative(final Name name) throws NamingException {
-        if (name.isEmpty()) {
```

Comment:
Again with empty appName part of the ejb address we are going to obtain WildflyRootContext instead of context provided for the ejb urlScheme which leads to an error.